From d050f796d61e755eb2e8b91434d09966fab92ff7 Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Sat, 10 Nov 2018 17:19:17 +0000
Subject: [PATCH 02/10] Telephony: Don not call onUssdRelease for Huawei RIL

Huawei RIL doesn't seem to work properly with USSD_MODE_NW_RELEASE,
always releasing USSD when it should be finished instead.
Let's explicitly call onUssdFinished in this case.

Change-Id: I69faed1c51d4582834879975d6ab13daf7f48ad4
---
 src/java/com/android/internal/telephony/GsmCdmaPhone.java | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index 93a0c2fd35..4cf72b5519 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -3138,7 +3138,11 @@ public class GsmCdmaPhone extends Phone {
         if (found != null) {
             // Complete pending USSD
             if (isUssdRelease) {
-                found.onUssdRelease();
+                if (SystemProperties.getBoolean("persist.sys.radio.huawei", false)) {
+                    found.onUssdFinished(ussdMessage, isUssdRequest);
+                } else {
+                    found.onUssdRelease();
+                }
             } else if (isUssdError) {
                 found.onUssdFinishedError();
             } else {
-- 
2.34.1

